src/Block.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Widgets;
+
+use Yiisoft\View\WebView;
+use Yiisoft\Widget\Widget;
+
+/**
+ * Block records all output between {@see begin()} and {@see end()} calls and stores it in
+ * {@see \Yiisoft\View\View::$blocks}.
+ *
+ * {@see \Yiisoft\View\View} component contains two methods {@see \Yiisoft\View\View::beginBlock()} and
+ * {@see \Yiisoft\View\View::endBlock()}.
+ *
+ * The general idea is that you're defining block default in a view or layout:
+ *
+ * ```php
+ * <?php $this->beginBlock('index') ?>
+ * Nothing.
+ * <?php $this->endBlock() ?>
+ * ```
+ *
+ * And then overriding default in views:
+ *
+ * ```php
+ * <?php $this->beginBlock('index') ?>
+ * Umm... hello?
+ * <?php $this->endBlock() ?>
+ * ```
+ *
+ * in subviews show block:
+ *
+ * <?= $this->getBlock('index') ?>
+ *
+ * Second parameter defines if block content should be outputted which is desired when rendering its content but isn't
+ * desired when redefining it in subviews.
+ */
+class Block extends Widget
+{
+    /**
+     * @var string $id
+     */
+    private string $id;
+
+    /**
+     * @var bool whether to render the block content in place. Defaults to false, meaning the captured block content
+     * will not be displayed.
+     */
+    private bool $renderInPlace = false;
+
+    private WebView $webView;
+
+    public function __construct(WebView $webview)
+    {
+        $this->webView = $webview;
+    }
+
+    /**
+     * Starts recording a block.
+     */
+    public function init(): void
```

Comment:
How about renaming it to `start()` or `capture()`?

Reply:
start(), done.